[4657] Reactify Global App (Sites & Users)

aachavarria opened the pull request:
https://github.com/craftercms/craftercms/issues/4657

aachavarria committed:
Initial UserGrid.tsx

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/global-menu/reactify

rart commented on ui/app/src/components/UsersGrid/UsersGrid.tsx:
```diff
@@ -0,0 +1,98 @@
+              </StyledTableCell>
+              <StyledTableCell align="left">
+                <Typography variant="subtitle2">
+                  <FormattedMessage id="words.userName" defaultMessage="Username" />
```
username (single word, non camelised)

rart commented on ui/app/src/components/UsersGrid/UsersGrid.tsx:
```diff
@@ -0,0 +1,98 @@
+            {users?.map((user) => (
+              <TableRow key={user.id}>
+                <TableCell padding="checkbox"></TableCell>
+                <TableCell component="th" id={user.id.toString()} scope="row" padding="none">
```
What’s the need for the id?

aachavarria replied:
not sure if the id is needed or why it is used, but in the material examples they are using the id and also using component="th"

rart commented on ui/app/src/components/UsersGrid/UsersGrid.tsx:
```diff
@@ -0,0 +1,98 @@
+  }, []);
+
+  return (
+    <ConditionalLoadingState isLoading={users === null}>
```
Why not use a resource?
Would be good to use a skeleton instead of loadingstate

aachavarria replied:
will change it to use resource and skeleton

rart commented on ui/app/src/components/UsersGrid/UsersGrid.tsx:
```diff
@@ -0,0 +1,98 @@
+export default function UsersGrid() {
+  const [users, setUsers] = useState<User[]>(null);
+
+  useEffect(() => {
```
Split into container and presentational components. Create separate fíles.

aachavarria committed:
Adding UserInfoDialog.tsx

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/global-menu/reactify

aachavarria committed:
UserInfoDialog.tsx: Adding Update Service;

aachavarria committed:
adding circular progress on save

rart commented on ui/app/src/components/UsersGrid/UserInfoDialog.tsx:
```diff
@@ -0,0 +1,278 @@
+  const [user, setUser] = useSpreadState(null);
+  const [lastSavedUser, setLastSavedUser] = useState(null);
+  const [inProgress, setInProgress] = useState(false);
+  const [dirty, setDirty] = useState(false);
```
Do you all need this state? i.e. after adding the save button

aachavarria replied:
yes because we need to keep the user being edited, and the original user, to do a rollback with the cancel button,
also the dirty helps to enable or disable the save button

rart commented on ui/app/src/components/UsersGrid/UsersGridUI.tsx:
```diff
@@ -0,0 +1,186 @@
+  })
+)(TableRow);
+
+interface UsersGridUIProps {
```
Export

rart commented on ui/app/src/components/UsersGrid/UsersGridUI.tsx:
```diff
@@ -0,0 +1,186 @@
+  );
+}
+
+interface UsersGridSkeletonTableProps {
```
Export and move to its own file together with component.
Remember to export from index too

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/global-menu/reactify

aachavarria committed:
Adding sites Memberships;
Updating EditMode to use ExternallyManaged;
UI Updates;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/global-menu/reactify

aachavarria committed:
Fixing issues when the dialog is closing

aachavarria committed:
Fetching sites roles;

aachavarria committed:
Creating Pagination Component;

aachavarria committed:
adding Pagination to UsersGrid.tsx (WIP)

aachavarria committed:
Fixing issue related with Fetch users Limit

aachavarria committed:
adding pagination support to UsersGrid.tsx;

aachavarria committed:
adding UsersGrid.tsx to GlobalMenu Users;
Adding Delete support to UsersInfoDialog.tsx;

aachavarria committed:
Cleaning old code;
Updating size on PrimaryButton <CircularProgress>;
Creating ResetPasswordDialog.tsx;

aachavarria committed:
Creating Sites Reactify Components;

aachavarria committed:
Cleaning up the code

aachavarria committed:
Moving Components to its own folders

aachavarria committed:
Adding Tooltip support to ConfirmDropdown.tsx;
Adding tooltips to UserInfoDialog.tsx;

aachavarria committed:
Making PasswordRequirementsDisplay a Component;
adding passwordRequirementsRegex to UsersGrid.tsx;

aachavarria committed:
LoginView.tsx: Updating to import PasswordRequirementsDisplay;
Fixing issues on some components exports

aachavarria committed:
Adding site delete support to SitesGrid.tsx;

aachavarria committed:
Removing mainContainer from GlobalMenu views;
UsersGrid.tsx: Adding EmptyState;
SitesGrid.tsx.tsx: Adding EmptyState and Pagination;

aachavarria committed:
adding default limit;

rart commented on static-assets/scripts/admin.js:
```diff
@@ -1194,69 +1194,14 @@
+        unmount();
+        CrafterCMSNext.render(document.getElementById('users-grid'), 'UsersGrid', {
+          passwordRequirementsRegex,
+          onClose: () => unmount()
```
Not applicable, this isn't a dialog

rart commented on ui/app/src/components/ResetPasswordDialog/ResetPasswordDialog.tsx:
```diff
@@ -0,0 +1,170 @@
+            values={{ user: props.user.username }}
+          />
+        </Typography>
+        <TextField
```
Use the password field component

rart commented on ui/app/src/components/ResetPasswordDialog/ResetPasswordDialog.tsx:
```diff
@@ -0,0 +1,170 @@
+        </SecondaryButton>
+        <PrimaryButton type="submit" onClick={onSubmit} autoFocus disabled={newPassword === '' || updating || !isValid}>
+          {updating ? <CircularProgress size={20} /> : <FormattedMessage id="words.submit" defaultMessage="Submit" />}
+        </PrimaryButton>
```
We need to make a component out of this. Too much repetition all over this PR

rart commented on ui/app/src/models/User.ts:
```diff
@@ -24,7 +24,7 @@ export interface User {
   enabled: boolean;
   externallyManaged: boolean;
-  authenticationType: 'db' | 'ldap' | 'headers' | 'saml';
+  authenticationType?: 'db' | 'ldap' | 'headers' | 'saml';
```
Why make this optional?

aachavarria committed:
Pagination: Moving interface to the top of de file;
SitesGrid: Using Sites from redux store;
SitesManagement: Integrating CreateSiteDialog;

aachavarria committed:
Integrating SitesManagement with Legacy;

aachavarria committed:
Creating ButtonWithLoadingState.tsx;
Updating Uses of ButtonWithLoadingState.tsx;

aachavarria committed:
adding fallbackImage to SiteCard.tsx;

rart commented on ui/app/src/components/ButtonWithLoadingState/ButtonWithLoadingState.tsx:
```diff
@@ -0,0 +1,34 @@
+  return (
+    <Button ref={ref} {...rest} variant="contained" color="primary">
+      {loading ? <CircularProgress size={20} /> : props.children}
+    </Button>
```
When loading, `disabled={true}` should be forced
```suggestion
    <Button ref={ref} {...rest} disabled={loading || rest.disabled} variant="contained" color="primary">
      {loading ? <CircularProgress size={20} /> : props.children}
    </Button>
```

rart commented on ui/app/src/components/UsersGrid/UsersGrid.tsx:
```diff
@@ -0,0 +1,129 @@
+/*
```
We should split this one up in the same way as sites: SitesManagement & SitesGrid

rart commented on ui/app/src/modules/System/Sites/Create/CreateSiteDialog.tsx:
```diff
@@ -293,7 +294,7 @@ const useStyles = makeStyles((theme: Theme) =>
 
 interface CreateSiteDialogProps {
-  // open: boolean;
+  open?: boolean;
```
Why? This should always be suppled to dialogs.

aachavarria committed:
PR Feedback;

aachavarria committed:
Creating UserDialog.tsx
Creating UsersManagement.tsx

aachavarria committed:
Fixing scroll issues on CreateUserDialog.tsx;

aachavarria committed:
Implementing Create Service on CreateUserDialog.tsx;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/global-menu/reactify

 Conflicts:
	ui/app/src/components/CreateTokenDialog/CreateTokenDialog.tsx

aachavarria committed:
Integrating UsersManagement.tsx to Legacy;

aachavarria committed:
Refactoring components;
Adding Grid/List view to SitesManagement.tsx;

aachavarria committed:
Refactoring on PublishingStatusAvatar

rart commented on ui/app/src/modules/System/Sites/Create/CreateSiteDialog.tsx:
```diff
@@ -303,7 +304,7 @@ function CreateSiteDialog(props: CreateSiteDialogProps) {
   const [disableEnforceFocus, setDisableEnforceFocus] = useState(false);
   const [dialog, setDialog] = useSpreadState({
-    open: true,
+    open: nnou(props.open) ? props.open : true,
```
This should be fully prop-driven. Ping me to discuss.
I'm guessing this has to do with the usage of the dialog in legacy, which is now not gonna happen.

aachavarria replied:
yes I need to update this logic

rart commented on ui/app/src/components/Controls/PasswordTextField.tsx:
```diff
@@ -34,7 +34,7 @@ const translations = defineMessages({
 });
 
-export default function PasswordTextField(props: PasswordTextFieldProps) {
+const PasswordTextField = React.forwardRef<JSX.Element, PasswordTextFieldProps>((props, ref) => {
```
You aren't propagating/using the ref on the body of the component? Also, the first generic type isn't JSX.Element, is whatever the component you give the ref to returns (i.e. HTMLDivElement, HTMLInputElement, or whatever)

aachavarria committed:
UsersGridSkeletonTable.tsx: Adding React Memo;
Moving UserGrid Logic to UsersManagement.tsx;
CreateUserDialog.tsx: Validation;

aachavarria committed:
Revert "Refactoring on PublishingStatusAvatar"

This reverts commit e2425fc6

aachavarria committed:
Merge branch '4.0.0/develop' of https://github.com/rart/studio-ui into feature/global-menu/reactify

 Conflicts:
	ui/app/src/components/PublishingStatusTile/PublishingStatusTile.tsx

aachavarria committed:
PasswordTextField.tsx: Updating Ref;
Adding PublishingStatusButton to SiteCard.tsx;
Fetching Sites Publishing status on SitesGrid.tsx;
UserInfoDialog.tsx: Fixing issues related to get my roles on each site;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/global-menu/reactify

aachavarria committed:
UserInfoDialog.tsx: Fixing issues related with Sites Roles;

aachavarria committed:
PublishingStatusDialogBody.tsx: Fixing issues Related to <div/> inside of <p></p>
Refactoring SiteGrid.tsx;

aachavarria committed:
Creating GroupsManagement.tsx;

rart commented on ui/app/src/components/PublishingStatusDialog/PublishingStatusDialogBody.tsx:
```diff
@@ -93,9 +93,9 @@ function PublishingStatusDialogBody(props: PublishingStatusDialogBodyProps) {
-                  {message && <div>{message}</div>}
-                  {lockOwner && <div>{formatMessage(publishingStatusTileMessages.lockOwner, { lockOwner })}</div>}
-                  {lockTTL && <div>{formatMessage(publishingStatusTileMessages.lockTTL, { lockTTL })}</div>}
+                  {message}
+                  {lockOwner && formatMessage(publishingStatusTileMessages.lockOwner, { lockOwner })}
+                  {lockTTL && formatMessage(publishingStatusTileMessages.lockTTL, { lockTTL })}
```
?

aachavarria replied:
Fixing a issue related with a <div> being child from <p>

rart replied:
This breaks the intended UI styling/layout. Can't just do this (test it).
What's the issue?

aachavarria committed:
Creating GlobalAppToolbar;
Updating UsersManagement.tsx, SitesManagement.tsx, TokenManagement.tsx to use GlobalAppToolbar.tsx
Creating GlobalAppGridRow and GlobalAppGridCell;
Updating Global Management tables to use GlobalAppGridRow and GlobalAppGridCell;

aachavarria committed:
Updating GroupsManagement.tsx placeholder;
